calculator.php

```diff
@@ -154,7 +148,7 @@ function show_history($history)
     if ($history === null) {
         info('You have no history');
     } else {
-        $historyGroups = array_group($history, 'date');
+        $historyGroups = array_group($history->savedData, 'date');
```

Comment:
all attributes should be protected inside the class, to get attribute value - please implement "getter" method

```suggestion
        $historyGroups = array_group($historyRepository->all(), 'date');
```